misc/xml-to-record-converter/src/main/java/io/ballerina/xmltorecordconverter/util/ConverterUtils.java

```diff
@@ -0,0 +1,181 @@
+        if (arrayNode.memberTypeDesc() instanceof ArrayTypeDescriptorNode) {
+            totalDimensions += getNumberOfDimensions((ArrayTypeDescriptorNode) arrayNode.memberTypeDesc());
+        }
+        return totalDimensions;
+    }
+
+    /**
+     * This method returns a list of TypeDescriptorNodes extracted from a UnionTypeDescriptorNode.
+     *
+     * @param typeDescNode UnionTypeDescriptorNode for which that has to be extracted.
+     * @return {@link List<TypeDescriptorNode>} The list of extracted TypeDescriptorNodes.
+     */
+    public static List<TypeDescriptorNode> extractUnionTypeDescNode(TypeDescriptorNode typeDescNode) {
+        List<TypeDescriptorNode> extractedTypeDescNodes = new ArrayList<>();
+        TypeDescriptorNode extractedTypeDescNode = typeDescNode;
+        if (typeDescNode.kind().equals(SyntaxKind.PARENTHESISED_TYPE_DESC)) {
+            extractedTypeDescNode = extractParenthesisedTypeDescNode(typeDescNode);
+        }
+        if (extractedTypeDescNode.kind().equals(SyntaxKind.UNION_TYPE_DESC)) {
+            UnionTypeDescriptorNode unionTypeDescNode = (UnionTypeDescriptorNode) extractedTypeDescNode;
+            TypeDescriptorNode leftTypeDescNode = unionTypeDescNode.leftTypeDesc();
+            TypeDescriptorNode rightTypeDescNode = unionTypeDescNode.rightTypeDesc();
+            extractedTypeDescNodes.addAll(extractUnionTypeDescNode(leftTypeDescNode));
+            extractedTypeDescNodes.addAll(extractUnionTypeDescNode(rightTypeDescNode));
+        } else {
+            extractedTypeDescNodes.add(extractedTypeDescNode);
+        }
+        return extractedTypeDescNodes;
+    }
+
+    private static TypeDescriptorNode extractParenthesisedTypeDescNode(TypeDescriptorNode typeDescNode) {
+        if (typeDescNode instanceof ParenthesisedTypeDescriptorNode) {
+            return extractParenthesisedTypeDescNode(((ParenthesisedTypeDescriptorNode) typeDescNode).typedesc());
+        } else {
+            return typeDescNode;
+        }
+    }
+
+    /**
+     * This method returns the SyntaxToken corresponding to the JsonPrimitive.
+     *
+     * @param value JsonPrimitive that has to be classified.
+     * @return {@link Token} Classified Syntax Token.
+     */
+    public static Token getPrimitiveTypeName(String value) {
+        if (isBoolean(value)) {
+            return AbstractNodeFactory.createToken(SyntaxKind.BOOLEAN_KEYWORD);
+        } else if (isInteger(value)) {
+            return AbstractNodeFactory.createToken(SyntaxKind.INT_KEYWORD);
+        } else if (isDouble(value)) {
+            return AbstractNodeFactory.createToken(SyntaxKind.DECIMAL_KEYWORD);
+        }
+        return AbstractNodeFactory.createToken(SyntaxKind.STRING_KEYWORD);
+    }
+
+    private static boolean isBoolean(String value) {
+        return value.equalsIgnoreCase("true") || value.equalsIgnoreCase("false");
+    }
+
+    private static boolean isInteger(String value) {
+        try {
+            Integer.parseInt(value);
```

Comment:
IMO we should check if the value is compatible with java long instead (as Ballerina `int` is represented as `long` datatype in runtime)

```suggestion
            Long.parseLong(value);
```